addons/stock_account/models/account_invoice.py

```diff
@@ -77,6 +77,56 @@ def _anglo_saxon_sale_move_lines(self, i_line):
                 ]
         return []
 
+    def _get_related_stock_moves(self):
+        """ To be overridden for customer invoices and vendor bills in order to
+        return the stock moves related to this invoice.
+        """
+        return self.env['stock.move']
+
+    def _get_products_set(self):
+        """ Returns a recordset of the products contained in this invoice's lines
+        """
+        return self.mapped('invoice_line_ids.product_id')
+
+    def _get_anglosaxon_interim_account(self, product):
+        """ Returns the interim account used in anglosaxon accounting for
+        this invoice"""
+        if self.type in ('out_invoice', 'out_refund'):
+            return product.product_tmpl_id._get_product_accounts()['stock_output']
+        elif self.type in ('in_invoice', 'in_refund'):
+            return product.product_tmpl_id.get_product_accounts()['stock_input']
+
+        return None
+
+    def invoice_validate(self):
+        super(AccountInvoice, self).invoice_validate()
+        self.anglo_saxon_reconcile_valuation(correct_invoice_interim_account=True)
+
+    def anglo_saxon_reconcile_valuation(self, correct_invoice_interim_account=False):
+        """ Reconciles the entries made in the interim accounts in anglosaxon accounting,
+        reconciling stock valuation move lines with the invoice's.
+        """
+        for invoice in self:
+            company_currency = invoice.company_id.currency_id
+            if invoice.company_id.anglo_saxon_accounting:
+                invoice_stock_moves = self._get_related_stock_moves()
+                all_stock_move_invoices = invoice_stock_moves._get_related_invoices()
+                for product in self._get_products_set():
+                    if product.valuation == 'real_time' and product.cost_method == 'fifo':
+                        # We first get the invoices move lines (taking the invoice and the previous ones into account)...
+                        product_interim_account = invoice._get_anglosaxon_interim_account(product)
+                        to_reconcile = self.env['account.move.line'].search([('move_id','in',all_stock_move_invoices.mapped('move_id.id')), ('product_id','=',product.id), ('account_id','=',product_interim_account.id), ('reconciled','=',False)])
+
+                        # And then the stock valuation ones.
+                        product_stock_moves = self.env['stock.move'].search([('id','in',invoice_stock_moves.ids), ('product_id','=',product.id)])
+
+                        for valuation_line in product_stock_moves.mapped('account_move_ids.line_ids'):
+                            if valuation_line.account_id == product_interim_account and not valuation_line.reconciled:
+                                to_reconcile += valuation_line
+
+                        if to_reconcile:
+                            to_reconcile.reconcile()
```

Comment:
Is there any provision to cancel the invoice after its Accrual (AngloSaxon) lines have been reconciled?

if trying to cancel an invoice where its lines have been reconciled Journal Entry/Reconciliation will complain.

https://github.com/Vauxoo/addons-vauxoo/blob/8.0/account_anglo_saxon_stock_move/model/invoice.py#L87-L90

Reply:
he got a point: that reconciliation on the interim account shouldn't prevent a cancellation of the invoice (but usually any reconciled journal item would)

Reply:
@oco-odoo & @qdp-odoo 
Will Invoice Cancellation be taken into account in this iteration?

Regards.